local/kaltura/yui/build/moodle-local_kaltura-ltipanel/moodle-local_kaltura-ltipanel-debug.js

```diff
@@ -285,11 +285,22 @@ Y.extend(LTIPANELMEDIAASSIGNMENT, Y.Base, {
      * @type {Object}
      */
     open_panel_callback : function(e, url, height, width) {
-        this.panelheight = parseInt(height, 10) + 45;
-        this.panelwidth = parseInt(width, 10) + 23;
+        var panelheight = parseInt(height, 10) + 45;
+        var panelwidth = parseInt(width, 10) + 23 + 'px';
+
+        width = width+'px';
+        // Apply special width for mobile devices as requested by Kaltura.
+        if (Y.UA.ipod !== 0 || Y.UA.ipad !== 0 || Y.UA.iphone !== 0 || Y.UA.android !== 0 || Y.UA.mobile !== null) {
+            panelwidth = '80%';
+            width = '100%';
+        }
 
-        this.panelbodycontent = "<iframe id='panelcontentframe' height='"+height+"px' width='"+width+"px' "+
-                "allowfullscreen='true' webkitallowfullscreen='true' mozallowfullscreen='true' src='"+url+"'></iframe>";
+        var iframe = "<iframe id='panelcontentframe' height='"+height+"px' width='"+width+"' src='"+url+"'></iframe>";
+        this.panelbodycontent = iframe;
+        if (Y.UA.ipod !== 0 || Y.UA.ipad !== 0 || Y.UA.iphone !== 0) {
+            // This outer div will constrain the iframe from overlapping over its content region on iOS devices.
+            this.panelbodycontent = "<div id='panelcontentframecontainer'>"+iframe+"</div>";
```

Comment:
did you make sure that this does not cause double rendering of the iframe?
i.e. when setting value to panelbodycontent - does that already affect the UI? or is UI updated at a later stage?